src/coreclr/System.Private.CoreLib/src/System/Enum.CoreCLR.cs

```diff
@@ -20,6 +20,13 @@ public abstract partial class Enum
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern unsafe CorElementType InternalGetCorElementType(MethodTable* pMT);
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static unsafe CorElementType InternalGetCorElementType(RuntimeType rt)
+        {
+            Debug.Assert(rt.IsActualEnum);
+            return InternalGetCorElementType((MethodTable*)rt.GetUnderlyingNativeHandle());
```

Comment:
Thanks.  I can measure, but theoretically why would this regress?  I thought the GC.KeepAlive would evaporate.  Is the concern this could inhibit inlining?

Reply:
Yes, GC.KeepAlive should evaporate when the optimizations are firing on all cylinders. I believe that we have seen cases in complicated code where it did not evaporate.